src/BingAdsEventForwarder.js

```diff
@@ -180,10 +220,126 @@ var constructor = function() {
+                var mpMappedConsentName = mappingEntry.map.toLowerCase();
+                var bingMappedConsentName = mappingEntry.value;
+
+                if (
+                    consentState[mpMappedConsentName] &&
+                    bingConsentProperties.indexOf(bingMappedConsentName) !== -1
+                ) {
+                    payload[bingMappedConsentName] = consentState[
+                        mpMappedConsentName
+                    ].Consented
+                        ? bingConsentValues.Granted
+                        : bingConsentValues.Denied;
+                }
+            }
+        }
+
+        return payload;
+    }
+
+    function maybeSendConsentUpdateToBing(consentState) {
+        // If consent payload is empty,
+        // we never sent an initial default consent state
+        // so we shouldn't send an update.
+        if (
+            self.consentPayloadAsString &&
+            self.consentMappings &&
+            !isEmpty(consentState)
+        ) {
+            var updatedConsentPayload = generateConsentPayload(
+                consentState,
+                self.consentMappings
+            );
+
+            var eventConsentAsString = JSON.stringify(updatedConsentPayload);
+
+            if (eventConsentAsString !== self.consentPayloadAsString) {
+                window.uetq.push('consent', 'update', updatedConsentPayload);
+                self.consentPayloadAsString = JSON.stringify(
+                    updatedConsentPayload
+                );
+            }
+        }
+    }
+
+    function sendConsentDefaultToBing(consentPayload) {
+        self.consentPayloadAsString = JSON.stringify(consentPayload);
+
+        window.uetq.push('consent', 'default', consentPayload);
+    }
+
     this.init = initForwarder;
     this.process = processEvent;
 };
 
+function getUserConsentState() {
+    var userConsentState = {};
+
+    if (mParticle.Identity && mParticle.Identity.getCurrentUser) {
+        var currentUser = mParticle.Identity.getCurrentUser();
+
+        if (!currentUser) {
+            return {};
+        }
+
+        var consentState = mParticle.Identity.getCurrentUser().getConsentState();
+
+        if (consentState && consentState.getGDPRConsentState) {
+            userConsentState = consentState.getGDPRConsentState();
+        }
+    }
+
+    return userConsentState;
+}
+
+function getConsentSettings(settings) {
+    var consentSettings = {};
+
+    Object.keys(bingToMpConsentSettingsMapping).forEach(function(
+        bingConsentKey
+    ) {
+        var mpConsentSettingKey =
+            bingToMpConsentSettingsMapping[bingConsentKey];
+        var bingConsentValuesKey = settings[mpConsentSettingKey];
+
+        // Microsoft recommends that for most countries, we should default to 'Granted'
+        // if a default value is not provided
+        // https://help.ads.microsoft.com/apex/index/3/en/60119
+        if (bingConsentValuesKey && mpConsentSettingKey) {
```

Comment:
I put this in as a safeguard in case the settings are not available, such as if the database record is ever unavailable.

I've done the same in [GTM](https://github.com/mparticle-integrations/mparticle-javascript-integration-google-tag-manager/blob/master/src/consent.js#L68), the only difference in this case is the fallback to `Granted`.